toolkit/tools/specreader/specreader.go

```diff
@@ -63,13 +66,23 @@ func main() {
 		logger.Log.Panicf("Value in --workers must be greater than zero. Found %d", *workers)
 	}
 
-	err := parseSPECsWrapper(*buildDir, *specsDir, *rpmsDir, *srpmsDir, *distTag, *output, *workerTar, *workers, *runCheck)
+	var toolchainRPMs []string
+	var err error
+	toolchainManifestPath := *toolchainManifest
+	if len(toolchainManifestPath) > 0 {
+		toolchainRPMs, err = schedulerutils.ReadToolchainPackageManifest(toolchainManifestPath)
+		if err != nil {
+			logger.Log.Fatalf("unable to read toolchain manifest file '%s': %s", toolchainManifestPath, err)
+		}
+	}
+
+	err = parseSPECsWrapper(*buildDir, *specsDir, *rpmsDir, *srpmsDir, toolchainRPMs, *existingToolchainRpmDir, *distTag, *output, *workerTar, *workers, *runCheck)
 	logger.PanicOnError(err)
 }
 
 // parseSPECsWrapper wraps parseSPECs to conditionally run it inside a chroot.
 // If workerTar is non-empty, parsing will occur inside a chroot, otherwise it will run on the host system.
-func parseSPECsWrapper(buildDir, specsDir, rpmsDir, srpmsDir, distTag, outputFile, workerTar string, workers int, runCheck bool) (err error) {
+func parseSPECsWrapper(buildDir, specsDir, rpmsDir, srpmsDir string, toolchainRPMs []string, toolchainDir, distTag, outputFile, workerTar string, workers int, runCheck bool) (err error) {
```

Comment:
No reason to repeat argument type for `string`:
```suggestion
func parseSPECsWrapper(buildDir, specsDir, rpmsDir, srpmsDir, toolchainDir, distTag, outputFile, workerTar string, toolchainRPMs []string, workers int, runCheck bool) (err error) {
```

Reply:
done